### DISCUSSION TITLE
How to refer to another file's frontmatter?

### DISCUSSION THREAD

**Discussion by ethan-harvey** [SilentVoid13/Templater#299] [Help]
If I'm running a template from one file, how can I get and use another file's frontmatter?

I want to create a list of files with headings that match the current file's title. Each of the files has the frontmatter key "time:" which I want to prepend to the file link and then use to sort the list. For example, if I'm running the template in a file named "2021-07-31.md", I want the list to look like this: 

```
- 11:00 [Event 1](event-1.md#2021-07-31)
- 20:00 [Event 2](event-2.md#2021-07-31)
```

I could use the Dataview plugin for this, but I'd like to be able to see the list in Edit mode and when I open the file with a different text editor.

(Alternatively, can I use the results of a Dataview query in a template?)

**Comment by AB1908** [comment]
You can use Dataview's API for this. Can also do in templates but would have to parse the file manually.

**Reply by wc8** [reply]
How would one parse it manually? 
I'd like to call from another page's field to fill in a link in a daily kanban, but the kanban isn't rendering the dataview inline correctly most of the time. Also, I'd like to have it dropped in once on the daily note, not changed later if I change the field.

**Reply by welpdx** [reply]
In [this dicussion showcase](https://github.com/SilentVoid13/Templater/discussions/174), @chhoumann lays the foundation for parsing and editing the yaml metadata using templater in obsidian.  He later made it into a plugin called [MetaEdit](https://github.com/chhoumann/MetaEdit). 

Recently, I saw a user called ryan [use the api from this metaedit plugin](https://github.com/SilentVoid13/Templater/discussions/533) for a quick and dirty way of parsing and editing the yaml metadata. 

Both are great resources to get your started to manually parse metadata for obsidian.

**Comment by ethan-harvey** [comment]
Somebody came up with a nice method for doing this here: https://github.com/SilentVoid13/Templater/discussions/1001
